book/src/04_4_1_interrupts.md

```diff
@@ -19,11 +19,18 @@ cargo run --example solution
     - See `hal::task::notification` documentation
 3. In an `unsafe` block, create a subscription and its callback function.
     - See `PinDriver::subscribe` and `task::notify_and_yield`
+    - The reasons for being `unsafe` are:
+      - The callback function will run in the [ISR (Interrupt Service Routine)](https://en.wikipedia.org/wiki/Interrupt_handler), so we should avoid calling any blocking functions on it, this includes STD, `libc` or FreeRTOS APIs (except for a few allowed ones).
+      - Callback closure is capturing its environment and you can use static variables inserted onto it. Captured variables need to outlive the subscription. You can also, use non-static variables, but that requires extra warning, see `esp_idf_hal::gpio::PinDriver::subscribe_nonstatic` documentation for more details.
```

Comment:
"requires extra caution" not warning maybe?

Reply:
Updated!